### ISSUE TITLE
sample_as_parent in variational_gamma

### ISSUE THREAD

**Issue by hyanwong** [tskit-dev/tsdate#394]
One of the tests for the inside_outside method is `test_sample_as_parent_fails`. Can we actually date samples that have descendant undated nodes using the current `variational_gamma` implementation, or should we be failing in this case too, @nspope ?

**Comment by nspope** [comment]
The machinery is (mostly) there but it's not working yet. Do we need to add an exception so that it errors out appropriately?

**Comment by hyanwong** [comment]
Right, I guess it would be helpful to error out, but it's not a high priority. Does the same go for non-contemporary samples?

**Comment by nspope** [comment]
Yup, same deal -- EP machinery in place, but rescaling routine doesn't do the right thing yet (although it's conceptually very straightforward to incorporate). Not going to worry about getting all that sorted out until we have a solid application.
